src/components/table/Table.tsx

```diff
@@ -1,23 +1,73 @@
-import { Table as AntdTable, TableProps } from 'antd'
+import { Table as AntdTable, SpinProps, TableProps } from 'antd'
 import cx from 'classnames'
 
 import s from './Table.module.css'
+import { Spinner } from '../spinner'
+import { CaretDown, CaretUp, CaretUpDown } from '@phosphor-icons/react'
 
 type MergedProps<T> = TableProps<T> & {
   verticalBorders?: boolean
+  columns: Exclude<TableProps<T>['columns'], undefined>
 }
 
 type Props<T> = Expand<MergedProps<T>>
 
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 const Table = <T = any,>({ verticalBorders, ...props }: Props<T>) => {
+  const defaultLoadingProps: SpinProps = {
+    indicator: <Spinner className="!w-fit !h-fit !mx-0 translate-x-[-50%]" />,
+    tip: (
+      <div className="mt-20">
+        <h4>Loading this chart</h4>
+        <span className="text-lg">Please wait a moment</span>
+      </div>
+    ),
+  }
+
+  const loading: SpinProps | boolean | undefined =
+    typeof props.loading == 'object'
+      ? {
+          ...props.loading,
+          ...defaultLoadingProps,
+        }
+      : props.loading && defaultLoadingProps
+
+  const columns = props.columns.map((column) => ({
+    ...column,
+    sortIcon: ({ sortOrder }: { sortOrder: 'ascend' | 'descend' | null }) => {
+      if (!column.sorter) {
+        return
+      }
+
+      if (sortOrder == 'ascend') {
+        return <CaretUp />
+      } else if (sortOrder == 'descend') {
+        return <CaretDown />
+      }
+
+      return <CaretUpDown />
+    },
```

Comment:
I also did not like doing this... the problem is that icons seem to be set at the column level instead of the table level, so I think this is the only way to get antd tables to use phosphor icons? But if we find a way we can specify just the up/down/updown carets somewhere else more efficiently then I'm all for that too